### ISSUE TITLE
Cloudflared package

### ISSUE THREAD

**Issue by Cadish** [mimugmail/opn-repo#14]
Hi, 

Would it be possible to add cloudflared as a package? 
https://github.com/cloudflare/cloudflared 

Thanks
Cadish

**Comment by mimugmail** [comment]
I asked over there about current pkg as the Port is outdated. Lets see :)

**Comment by Cadish** [comment]
Great! Thanks!

**Comment by Cadish** [comment]
hi @mimugmail , they have merged the code apparently... https://github.com/cloudflare/cloudflared/issues/285

**Comment by mimugmail** [comment]
No, the author merged it into his own fork, it's still open.

**Comment by mimugmail** [comment]
It's now in community repo:

`pkg install cloudflared
`
If you tell me something about config and syntax I can try to build a plugin

**Comment by Cadish** [comment]
Great! Will try it. Thanks a lot. 

The config & syntax are dependent on what you want to achieve with it. I only use it with Cloudflare Argo to create a tunnel to disclose internal applications to the outside without opening any ports. This way, I also can use Cloudflare Access as an extra security level to not disclose my origins directly. 

To do this:
1. Create a tunnel itself: `cloudflared tunnel create <NAME>`
2. Create a config file: https://developers.cloudflare.com/cloudflare-one/connections/connect-apps/configuration/config
3. Run the tunnel: `cloudflared tunnel --config path/config.yaml run <NAME>`.

More documentation on this is here: https://developers.cloudflare.com/cloudflare-one/connections/connect-apps

**Comment by mimugmail** [comment]
If you got it running on the OPN you can also send me the config via private channel and I can try to build a plugin for you.

**Comment by MattElekHarris** [comment]
I'd love to see support for this, especially considering that [Cloudflare Tunnel is free now](https://blog.cloudflare.com/tunnel-for-everyone/). I imagine some people might like to use it for DNS over HTTPS, although dnscrypt-proxy already exists in the main repository.

**Comment by hobbes444** [comment]
Am I reading it correctly and it's now included in 2023.5.0? I'm a bit confused because it's still open...
https://github.com/cloudflare/cloudflared/issues/285

**Comment by hobbes444** [comment]
why doesn't `cloudflared service install` work? That would be so much simpler, single command to set up the tunnel.

**Comment by sammyke007** [comment]
Did anyone manage to run this as a service?

**Comment by Qhilm** [comment]
> Did anyone manage to run this as a service?

This worked for me:

https://www.jackpearce.co.uk/cloudflared-opnsense/

**Comment by sammyke007** [comment]
> > Did anyone manage to run this as a service?
> 
> This worked for me:
> 
> https://www.jackpearce.co.uk/cloudflared-opnsense/

Yes, that's exactly what I use, but Cloudflared's tunnel as a plugin (service) would be even better.

**Comment by bobpaul** [comment]
Hmm, I tried using the package as-is (following cloudflare's instructions for a locally managed tunnel) and things seemed to go OK (created cert.pem, created config.yml + credentials.json, etc) but then when I tried to start the connection I got

```
2025-06-20T22:40:29Z ERR Failed to dial a quic connection error="failed to dial to edge with quic: INTERNAL_ERROR (local): tls: CurvePreferences includes unsupported curve" connIndex=0 event=0 ip=
```

So I tried following Jack Pearce's instructions (create tunnel from website, edit the rc script, put the token in rc.conf) and that has the same problem.

I found [others reporting this issue](https://github.com/cloudflare/cloudflared/issues/1148) and it sounds like the package needed to be rebuilt to fix it.

I'm on `stable/25.1-n269825-cbaf3aa6b26b SMP amd64`

**Comment by Qhilm** [comment]
Same issue for me, it's broken. Not sure exactly when it started

`2025-06-22T14:51:59Z ERR Failed to dial a quic connection error="failed to dial to edge with quic: INTERNAL_ERROR (local): tls: CurvePreferences includes unsupported curve" connIndex=0 event=0 ip=redacted`

This is the command I have been using to start cloudflared after each restart of opnsense (I never got it to autostart):

`/usr/sbin/daemon -o /var/log/cloudflared.log -p /var/run/cloudflared.pid -f /usr/local/bin/cloudflared tunnel --no-autoupdate run --post-quantum --token xxx`

I'm running cloudflared 2024.11.1.

**Comment by bobpaul** [comment]
@Qhilm I found a [thread about it](https://forum.opnsense.org/index.php?topic=42168.msg239157#msg239157) on the opnsense forums. It works if I build it from ports using `go124` _AND_ disable the `--post-quantum` option.

**Comment by bobpaul** [comment]
> If you got it running on the OPN you can also send me the config via private channel and I can try to build a plugin for you.

I think we need to decide what we want the plugin to be capable of: device managed tunnels or remotely managed tunnels.

It looks like using the `--token` option, as recommended by JackPearce allows configuration of the tunnel endpoints remotely (from the cloudflare website), which a lot of users might prefer. But as there's no "token" option in the config file, so this can't be used with the `--config` option provided by the rc file included in the package.

Using the config file is a quite a bit more work (login or import API key, create tunnel or import tunnel credentials, create dns entries and route to tunnels, manage config.yml to map routes to local services). Using the config file moves all of the configuration complexity to the plugin. 

Having set up tunnels both ways, I think I would recommend the plugin only supports the `--token` method. The plugin will only need to ask for the tunnel token and provide options to start, stop, restart the tunnel. The plugin could also provide some instruction to create/manage the tunnel on https://one.dash.cloudflare.com

Technically one could run multiple copies of cloudflared (each with a different token). The main reason for wanting to do this would be someone with multiple cloudflare accounts.
